apps/andi/test/unit/andi_web/live/edit_live_view_test.exs

```diff
@@ -149,126 +162,143 @@ defmodule AndiWeb.EditLiveViewTest do
+      dataset = DatasetHelpers.create_dataset(%{})
+
+      DatasetHelpers.add_dataset_to_repo(dataset)
+
       expected = Enum.join(dataset.business.keywords, ", ")
 
-      dataset_map =
-        dataset
-        |> dataset_to_form_data()
-        |> Map.put(:keywords, expected)
+      form_data =
+        FormTools.form_data_from_andi_dataset(dataset)
+        |> put_in([:business, :keywords], expected)
 
       assert {:ok, view, _html} = live(conn, @url_path <> dataset.id)
-      html = render_change(view, :validate, %{"form_data" => dataset_map})
+      html = render_change(view, :validate, %{"form_data" => form_data})
 
-      subject = get_value(html, "#form_data_keywords")
+      subject = get_value(html, ".metadata-form__keywords input")
 
       assert expected == subject
     end
 
     test "displays all other fields", %{conn: conn} do
       dataset =
-        TDG.create_dataset(%{
-          business: %{description: "A description with no special characters", benefitRating: 1.0, riskRating: 0.5},
+        DatasetHelpers.create_dataset(%{
+          business: %{
+            description: "A description with no special characters",
+            benefitRating: 1.0,
+            riskRating: 0.5
+          },
           technical: %{private: true}
         })
 
-      DatasetCache.put(dataset)
+      DatasetHelpers.add_dataset_to_repo(dataset)
 
       assert {:ok, _view, html} = live(conn, @url_path <> dataset.id)
-      assert get_value(html, "#form_data_dataTitle") == dataset.business.dataTitle
-      assert get_text(html, "#form_data_description") == dataset.business.description
-      assert get_value(html, "#form_data_sourceFormat") == dataset.technical.sourceFormat
-      assert {"true", "Private"} == get_select(html, "#form_data_private")
-      assert get_value(html, "#form_data_contactName") == dataset.business.contactName
-      assert dataset.business.modifiedDate =~ get_value(html, "#form_data_modifiedDate")
-      assert get_value(html, "#form_data_contactEmail") == dataset.business.contactEmail
-      assert dataset.business.issuedDate =~ get_value(html, "#form_data_issuedDate")
-      assert get_value(html, "#form_data_license") == dataset.business.license
-      assert get_value(html, "#form_data_publishFrequency") == dataset.business.publishFrequency
-      assert get_value(html, "#form_data_spatial") == dataset.business.spatial
-      assert get_value(html, "#form_data_temporal") == dataset.business.temporal
-      assert get_value(html, "#form_data_orgTitle") == dataset.business.orgTitle
-      assert {"english", "English"} == get_select(html, "#form_data_language")
-      assert get_value(html, "#form_data_homepage") == dataset.business.homepage
-      assert {"1.0", "High"} == get_select(html, "#form_data_benefitRating")
-      assert {"0.5", "Medium"} == get_select(html, "#form_data_riskRating")
+      assert get_value(html, ".metadata-form__title input") == dataset.business.dataTitle
```

Comment:
would it be cleaner to store these selectors in an external file and import them when you need them.

Reply:
Eh I don't know, I think they are helpful in the assertions to tell which piece of the view we are looking at